### ISSUE TITLE
AzureAD 2.x groups incompatible with AzureAD 3.x

### ISSUE THREAD

**Issue by shawnjones253** [pulumi/pulumi-azuread#147]
TLDR; does auto-naming no longer work with AzureAD 3.0?

Groups created with AzureAD < 3.0 can't easily be managed with AzureAD >= 3.0. Here's an example:

```
// with AzureAD v 2.6.0 or lower
var group = new Group("my-group", new GroupArgs()
{
    Description = "description-of-group"
});
// creates a group that's actually named something like "my-groupb3e7f11"
```

```
// with AzureAD v 3.0.0 or higher
var group = new Group("my-group", new GroupArgs()
{
    Description = "description-of-group"
});
```

pulumi up then gives you this error:
```
error: azuread:index/group:Group resource 'my-group' has a problem: ExactlyOne: "display_name": one of `display_name,name` must be specified
    error: azuread:index/group:Group resource 'my-group' has a problem: ExactlyOne: "name": one of `display_name,name` must be specified
```

so I try to specify the exact group name:
```
var group = new Group("my-group", new GroupArgs()
{
    DisplayName = "my-groupb3e7f11",
    Description = "description-of-group"
});
```

pulumi up then wants to replace the existing group:
```
++azuread:index/group:Group: (create-replacement)
        [id=....]
        [urn=urn:pulumi:stackName::stack::azuread:index/group:Group::my-group]
        [provider: urn:pulumi:stackName::stack::pulumi:providers:azuread::default_2_4_0::0000000 => urn:pulumi:stackName::stack::pulumi:providers:azuread::default_3_4_0::output<string>]
      + displayName: "my-groupb3e7f11"
```

trying again with `Name` instead:
```
var group = new Group("my-group", new GroupArgs()
{
    Name = "my-groupb3e7f11",
    Description = "description-of-group"
});
```

and finally that works, but:
- now i'm stuck with the hardcoded group name, so auto naming doesn't work
- I also get a warning that I'm using a deprecated property
```
azuread:index:Group        my-group                                                 [diff: ~__defaults,provider]; 1 warning

azuread:index:Group (my-group):
    warning: urn:pulumi:stackName::stack::azuread:index/group:Group::my-group verification warning: Deprecated Attribute
```

---

Additionally, I tried using `pulumi import` to see exactly what code I should be using, but it breaks in the same way:
```
azuread:index:Group (my_group):
    error: azuread:index/group:Group resource 'my_group' has a problem: ExactlyOne: "name": one of `display_name,name` must be specified
    error: azuread:index/group:Group resource 'my_group' has a problem: ExactlyOne: "display_name": one of `display_name,name` must be specified
```

**Comment by matteocalabro-tomtom** [comment]
Having a similar issue: `pulumi import` is broken in 3.5.0:

```
❯ pulumi import azuread:index/group:Group a_group abb1fcbe-6424-4001-b224-007356a15ffb
Previewing import (groups):
     Type                    Name                                     Plan       Info
 +   pulumi:pulumi:Stack     azuread-groups                     create     1 error
 =   └─ azuread:index:Group  a_group  import     3 errors

Diagnostics:
  pulumi:pulumi:Stack (azuread-groups):
    error: preview failed

  azuread:index:Group (mygroup):
    error: azuread:index/group:Group resource 'a_group' has a problem: ExactlyOne: "display_name": one of `display_name,name` must be specified
    error: azuread:index/group:Group resource 'a_group' has a problem: ExactlyOne: "name": one of `display_name,name` must be specified
    error: Preview failed: one or more inputs failed to validate
```

even with a fresh stack. It's currently impossible to import groups (I haven't tried the JSON-file way)

**Comment by guineveresaenger** [comment]
Hi everyone! I've run through a few scenarios here. Apologies that this issue is stale; however the good news is that it is possible to successfully import Groups using the most recent AzureAD provider, meaning that this issue can be resolved by upgrading if you have not done so already.

Caveat: It is not trivial to use v2 of this provider because of [authentication upgrades in the Azure CLI](https://github.com/Azure/azure-cli/issues/20356). I did, however, manage to duplicate most of this behavior between v3.0.0 and v 3.5.0.

**Create a Group using v3.0.0; upgrade to 3.5.0, attempt pulumi import.**
Preview output:
```
pulumi import azuread:index/group:Group example-c451e72 79d1ce6d-0cb6-4acf-bf61-5206cf296ab3

     Type                    Name             Plan       Info
     pulumi:pulumi:Stack     azuread-147-dev             
 =   └─ azuread:index:Group  example-c451e72  import     3 warnings


Diagnostics:
  azuread:index:Group (example-c451e72):
    warning: One or more imported inputs failed to validate. This is almost certainly a bug in the `azuread` provider. The import will still proceed, but you will need to edit the generated code after copying it into your program.
    warning: azuread:index/group:Group resource 'example-c451e72' has a problem: ExactlyOne: "display_name": only one of `display_name,name` can be specified, but `display_name,name` were specified.
    warning: azuread:index/group:Group resource 'example-c451e72' has a problem: ExactlyOne: "name": only one of `display_name,name` can be specified, but `display_name,name` were specified.
```

**Upgrade to v5.40.0 of the pulumi-azuread provider; attempt import again**

Import succeeds on preview, and provides us with upgraded code for the new shape of the resource:

```
const example_c451e72 = new azuread.Group("example-c451e72", {
    description: "example group",
    displayName: "example-c451e72",
    mailNickname: "ff12be0e-4e2b-0e52-b86d-d8fca8707320",
    owners: ["xxx],
    preventDuplicateNames: false,
    securityEnabled: true,
    writebackEnabled: false,
}, {
    protect: true,
});
```

The import in this case does require the code to use the name `example-c451e72`, which is because v3.0.0 created it with that reference.

Please let us know if this continues to be an issue on current versions of this provider.

I'm closing this issue for now.
